File: mambo-shared/mambo-protocol/mambo-protocol-client/src/main/java/org/mambo/protocol/client/messages/MapRunningFightDetailsMessage.java


// Generated on 12/14/2012 18:44:07
package org.mambo.protocol.client.messages;

import java.util.*;
import org.mambo.protocol.client.types.*;
import org.mambo.protocol.client.enums.*;
import org.mambo.protocol.client.*;
import org.mambo.core.io.*;

public class MapRunningFightDetailsMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5751;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int fightId;
    public String[] names;
    public short[] levels;
    public byte teamSwap;
    public boolean[] alives;
    
    public MapRunningFightDetailsMessage() { }
    
    public MapRunningFightDetailsMessage(int fightId, String[] names, short[] levels, byte teamSwap, boolean[] alives) {
        this.fightId = fightId;
        this.names = names;
        this.levels = levels;
        this.teamSwap = teamSwap;
        this.alives = alives;
    }
    
    @Override
    public void serialize(DataWriterInterface writer) {
        writer.writeInt(fightId);
        writer.writeUnsignedShort(names.length);
        for (String entry : names) {
            writer.writeString(entry);
        }
        writer.writeUnsignedShort(levels.length);
        for (short entry : levels) {
            writer.writeShort(entry);
        }
        writer.writeByte(teamSwap);
        writer.writeUnsignedShort(alives.length);
        for (boolean entry : alives) {
            writer.writeBoolean(entry);
        }
    }
    
    @Override
    public void deserialize(DataReaderInterface reader) {
        fightId = reader.readInt();
        if (fightId < 0)
            throw new RuntimeException("Forbidden value on fightId = " + fightId + ", it doesn't respect the following condition : fightId < 0");
        int limit = reader.readUnsignedShort();
        names = new String[limit];
        for (int i = 0; i < limit; i++) {
            names[i] = reader.readString();
        }
        limit = reader.readUnsignedShort();
        levels = new short[limit];
        for (int i = 0; i < limit; i++) {
            levels[i] = reader.readShort();
        }
        teamSwap = reader.readByte();
        if (teamSwap < 0)
            throw new RuntimeException("Forbidden value on teamSwap = " + teamSwap + ", it doesn't respect the following condition : teamSwap < 0");
        limit = reader.readUnsignedShort();
        alives = new boolean[limit];
        for (int i = 0; i < limit; i++) {
            alives[i] = reader.readBoolean();
        }
    }
    
}

File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/MapRunningFightDetailsMessage.scala





















// Generated on 05/03/2013 10:34:42
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait MapRunningFightDetailsMessage extends MessageDeserializer {
  val messageId = 5751

  protected trait Message extends BaseMessage with Serializable {
    val messageId = MapRunningFightDetailsMessage.messageId

    val fightId: Int
    val names: Seq[String]
    val levels: Seq[Short]
    val teamSwap: Byte
    val alives: Seq[Boolean]

    def serialize(buf: Buffer) {
      buf.writeInt(fightId)
      buf.writeUShort(names.length)
      for (entry <- names) {
        buf.writeString(entry)
      }
      buf.writeUShort(levels.length)
      for (entry <- levels) {
        buf.writeShort(entry)
      }
      buf.writeByte(teamSwap)
      buf.writeUShort(alives.length)
      for (entry <- alives) {
        buf.writeBoolean(entry)
      }
    }

  }

  protected def create(fightId: Int, names: Seq[String], levels: Seq[Short], teamSwap: Byte, alives: Seq[Boolean]): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val fightId = buf.readInt()
    val names = {
      val builder = collection.immutable.Seq.newBuilder[String]
      for (i <- (0 to buf.readUShort())) {
        builder += buf.readString()
      }
      builder.result()
    }
    val levels = {
      val builder = collection.immutable.Seq.newBuilder[Short]
      for (i <- (0 to buf.readUShort())) {
        builder += buf.readShort()
      }
      builder.result()
    }
    val teamSwap = buf.readByte()
    val alives = {
      val builder = collection.immutable.Seq.newBuilder[Boolean]
      for (i <- (0 to buf.readUShort())) {
        builder += buf.readBoolean()
      }
      builder.result()
    }

    create(fightId, names, levels, teamSwap, alives)
  }

}

object MapRunningFightDetailsMessage extends MapRunningFightDetailsMessage {
  case class MessageImpl(fightId: Int, names: Seq[String], levels: Seq[Short], teamSwap: Byte, alives: Seq[Boolean]) extends Message
  type Target = MessageImpl

  override def create(fightId: Int, names: Seq[String], levels: Seq[Short], teamSwap: Byte, alives: Seq[Boolean]) = MessageImpl(fightId, names, levels, teamSwap, alives)

  def apply(fightId: Int, names: Seq[String], levels: Seq[Short], teamSwap: Byte, alives: Seq[Boolean]) = create(fightId, names, levels, teamSwap, alives)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

